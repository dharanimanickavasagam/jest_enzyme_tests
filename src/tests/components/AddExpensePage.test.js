import { AddExpensePage } from "./../../components/AddExpensePage";
import React from "react";
import { shallow } from "enzyme";
import expenses from "../fixtures/expenses";

let onSubmitSpy, historySpy, wrapper;
beforeEach(() => {
  onSubmitSpy = jest.fn();
  historySpy = { push: jest.fn() };
  wrapper = shallow(
    <AddExpensePage onSubmit={onSubmitSpy} history={historySpy} />
  );
});

describe("Should render the component", () => {
  test("add an expense", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("add an expense", () => {
    wrapper.find("ExpenseForm").prop("onSubmit")(expenses[1]);
    expect(historySpy.push).toHaveBeenLastCalledWith("/");
    expect(onSubmitSpy).toHaveBeenLastCalledWith(expenses[1]);
  });
});
