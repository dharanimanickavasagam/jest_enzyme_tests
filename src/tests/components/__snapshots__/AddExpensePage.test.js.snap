// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the component add an expense 1`] = `
<div>
  <h1>
    Add Expense
  </h1>
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
